lib/mini_twitter_client.rb

```diff
@@ -0,0 +1,63 @@
+require 'faraday'
+require 'json'
+
+class MiniTwitterClient
+  def connection
+    Faraday.new(
+      url: 'http://localhost:4000/api/tweets',
+      params: {},
+      headers: {
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/json'
+      }
+    )
+  end
```

Comment:
Each time you call the `connection` method, the above code is executed, and as a result you create a new instance of Faraday class (you call `Faraday.new`).

For example, if I execute the following code:

```ruby
client = MiniTwitterClient.new
client.get_tweets
client.get_tweets
```

2 instances on `Faraday.new` would be created in the memory.

What you can do is the following:

You can create it once when Client is instantiated:

```ruby
def initialize
  @connection = Faraday.new(....)
end
```

Then you could use single `@connection` inside this class.

---

The other benefit is that this would stop working:

```ruby
client = MiniTwitterClient.new
connection = client.connection
```

In the other words, the connection would be hidden inside the client, and would not leak outside.